File: 프로그래머스/unrated/181916. 주사위 게임 3/주사위 게임 3.java
import java.util.Arrays;
class Solution {
    public int solution(int a, int b, int c, int d) {
        int answer = 0;
        int NumArr[] = {a, b, c, d};
        Arrays.sort(NumArr);
        
        if(NumArr[0] == NumArr[3]) return 1111 * NumArr[0];
        else if(NumArr[0] == NumArr[2] || NumArr[1] == NumArr[3]) return (int)Math.pow((NumArr[1] * 10)  + NumArr[0] + NumArr[3] - NumArr[1], 2);
        else if(NumArr[0] == NumArr[1] && NumArr[2] == NumArr[3]) return (NumArr[0] + NumArr[3]) * (NumArr[3] - NumArr[0]);
        else if(NumArr[0] == NumArr[1]) return NumArr[2] * NumArr[3];
        else if(NumArr[1] == NumArr[2]) return NumArr[0] * NumArr[3];
        else if(NumArr[2] == NumArr[3]) return NumArr[0] * NumArr[1];
        else return NumArr[0];

    }
}

File: 프로그래머스/0/181916. 주사위 게임 3/주사위 게임 3.js
function solution(a, b, c, d) {
    var arr = [a, b, c, d];
    arr.sort();
    
    if(arr[0] == arr[3]) return 1111 * arr[0];
    else if(arr[0] == arr[2] || arr[1] == arr[3]) return Number(Math.pow((arr[1] * 10)  + arr[0] + arr[3] - arr[1], 2));
    else if(arr[0] == arr[1] && arr[2] == arr[3]) return (arr[0] + arr[2]) * (arr[2] - arr[0]);
    else if(arr[0] == arr[1]) return arr[2] * arr[3];
    else if(arr[1] == arr[2]) return arr[0] * arr[3];
    else if(arr[2] == arr[3]) return arr[0] * arr[1];
    else return arr[0];
}